stellar-contract-env-host/src/host.rs

```diff
@@ -597,11 +597,18 @@ impl CheckedEnv for Host {
     }
 
     fn map_put(&self, m: Object, k: RawVal, v: RawVal) -> Result<Object, HostError> {
-        todo!()
+        let k = self.associate_raw_val(k);
+        let v = self.associate_raw_val(v);
+        let mnew = self.visit_obj(m, |hm: &HostMap| Ok(hm.update(k, v)))?;
```

Comment:
Cool, I didn't see the `update` function and used `insert` instead. This looks much better.

Reply:
Ah interesting, I didn't see insert. It looks like they do almost the same thing. The API for these collection types are interesting and not completely intuitive to me. i.e. I find it confusing that 'update' returns a copy and doesn't actually update the existing map, whilst 'insert' does. 🤷🏻

Reply:
Haha my thoughts exactly.

Reply:
Yeah, `update` is just a `clone + insert`: https://docs.rs/im/latest/src/im/ord/map.rs.html#832-836